### PR TITLE
Oy2 23387 - display last rai response date in package table and details page for CMS users

### PR DESCRIPTION
**Pull request by bflynn-collabralink**
Story: https://qmacbis.atlassian.net/browse/OY2-23387
Endpoint: See github-actions bot comment

### Details

Add "Formal RAI Received" column to package view and as a field on detail view. This is only visible by CMS users.

### Changes

- Add column to Waiver Package Table for CMS Users
- Add field to details page for CMS Users
- Updated buildAnyPackage to set latestRaiResponseTimestamp when an RAI is submitted
- Updated migrate.js to update all existing packages to have new latestRaiResponseTimestamp field
- reworked how waiver detail default fields were configured

### Implementation Notes

- Added new property to DetailAttribute that allows specification of a UserRole property (privilege) by which the currently logged in user must have in order for the field to be displayed

### Test Plan

1. Login as CMS user
2. Verify "Formal RAI Response" exists in package view on waivers tab as a column
3. Locate package that has an RAI Response and verify value in column matches latest RAI Response date
4. Navigate to Details page for a package with an RAI Response and verify value of "Formal RAI Response" matches latest RAI Respsonse date
5. Login as State user
6. Verify column does not exist in package view on waivers tab or on details page
7. Locate a package in RAI Issued status and submit an RAI Response
8. Login as CMS User and verify current date appears in "Formal RAI Response" column and field

### CODE REVIEW

**Review by kristin-at-theta — CHANGES_REQUESTED** [review]
I love the consolidation of the waiver defaults!

Please fix that the column is on both SPA and Waiver tabs.

Please move that map out of every event.

Please update the migrate so it does Both the TE and the RAI updates :)

File: services/one-stream/package/buildAnyPackage.js [hunk 1]
```diff
@@ -107,6 +107,9 @@ export const buildAnyPackage = async (packageId, config) => {
           additionalInformation: anEvent.additionalInformation,
         });
         putParams.Item.currentStatus = Workflow.ONEMAC_STATUS.SUBMITTED;
+        putParams.Item.latestRaiResponseTimestamp = Math.max(
```
[the remark this answers is not in the document]
Comment (bflynn-collabralink) [reply services/one-stream/package/buildAnyPackage.js:110]: indeed, didnt catch that we were already sorting them later on. Just seemed natural to put the logic with the other RAI processing that I did see. :)
Will refactor.

File: services/app-api/migrate.js [hunk 2]
```diff
@@ -48,5 +48,70 @@ export const main = handler(async () => {
...
+      console.log("updating: ", anUpdate);
+      await dynamoDb.update(anUpdate);
+    })
+  );
+
+  console.log("migrateRaiTimestamp complete");
+};
+
+export const main = handler(async () => {
+  await migrateTE();
```
Comment (kristin-at-theta) [inline services/app-api/migrate.js:114]: Thank you!!!  Actually, this looks like a nice pattern to move forward with!  Great job :)

**Review by kristin-at-theta — APPROVED** [review]
Tests well, sorts, results export with correct looking date, added a SPA RAI Response to check and it also worked!

Thanks for making the changes :) Approved!

**Pull request merged**